### ISSUE TITLE
Gaze Cursor's SetVisibility() not working

### ISSUE THREAD

**Issue by Pycorax** [microsoft/MixedRealityToolkit-Unity#5395]
## Bug Description
`MixedRealityToolkit.InputSystem.EyeGazeProvider.GazeCursor.SetVisibility()` does not do anything.

## To Reproduce

1. Call `MixedRealityToolkit.InputSystem.EyeGazeProvider.GazeCursor.SetVisibility(false);`,
2. Observe the cursor not turning invisible

## Expected behavior
Upon calling `MixedRealityToolkit.InputSystem.EyeGazeProvider.GazeCursor.SetVisibility(false);`, the cursor should turn invisible.

## Setup

- Unity Version 2018.40f1
- MRTK Version 2.0.0 RC2.1

## Target Platform
- HoloLens

## Additional context
Calling `MixedRealityToolkit.InputSystem.EyeGazeProvider.GazeCursor.GameObjectReference.SetActive(false);` seems to work as a temporary workaround to achieve the desired effect of hiding the cursor.

**Comment by michel-de-verdelhan** [comment]
I have the same problem. I found that the Gaze cursor visibility is actually driven by 
FocusProvider.ReconcilePointers() which re enable it based on it's internal state machine. I think adding an option to disable this behaviour would be helpful

**Comment by timGerken** [comment]
@wiwei, is there a way to work around this in the 2.0.0 release without waiting for 2.0.1?

In 2.0.0-RC2.1 I was able to call `inputSystem.GazeProvider.Enabled = state;` to effectivly show/hide the gaze cursor, but when moving to 2.0.0 GA, apparently the FocusProvider has a dependency on the GazeProvider being enabled, and leads to the following exception:
```
NullReferenceException: Object reference not set to an instance of an object
Microsoft.MixedReality.Toolkit.Input.FocusProvider.UpdatePointer (Microsoft.MixedReality.Toolkit.Input.FocusProvider+PointerData pointerData) (at Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs:903)
Microsoft.MixedReality.Toolkit.Input.FocusProvider.UpdatePointers () (at Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs:840)
Microsoft.MixedReality.Toolkit.Input.FocusProvider.Update () (at Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs:517)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit+<>c.<UpdateAllServices>b__60_0 (Microsoft.MixedReality.Toolkit.IMixedRealityService service) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:883)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.ExecuteOnAllServices (System.Collections.Generic.IEnumerable`1[T] services, System.Action`1[T] execute) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:972)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.ExecuteOnAllServicesInOrder (System.Action`1[T] execute) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:953)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.UpdateAllServices () (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:883)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.Update () (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:583)
```

**Comment by wiwei** [comment]
@Pycorax, @timGerken FYI, I chatted with @julenka offline about the linked PR, and we decided it would make sense to roll up the solution to this issue into another issue that she is tracking.

https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4954

The main thing here is that there are a few other scenarios where people want to disable/enable things at runtime (i.e. turn of hand ray, turn off poke pointer, turn off gaze).

The key thing there is that with that change, we're probably going to be disabling the gaze pointer as a whole, rather than just the cursor - if this isn't what you want (i.e. you actually want the gaze pointer to still be active, but just hide the cursor so it doesn't look like it's active) then please comment on https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4954 so that we can capture that appropriately.

In the interim to unblock yourself, you can definitely patch https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5799 that locally to get the desired behavior.

**Comment by wiwei** [comment]
I think in my approach to solving this issue, my focus was too narrow, and I do think it makes sense to approach this more holistically across other input mechanisms.
